frontend/src/scenes/AdminPage.tsx

```diff
@@ -90,117 +86,144 @@ function AllowlistTabContent() {
-          }}
-        />
-      )}
-      <div style={{ display: 'flex', flexDirection: 'column' }}>
-        <InputLabel id="allowlist-country-label">Country</InputLabel>
-        <Select
-          style={{ background: 'white' }}
-          labelId="allowlist-country-label"
-          onChange={(e: React.ChangeEvent<any>) =>
-            setNewAllowlistCountry(e.target.value)
-          }
+    <form onSubmit={onSubmit}>
+      <Grid
+        container
+        alignItems="center"
+        spacing={2}
+        style={{ margin: '8px 0 0 0' }}
+      >
+        {useMultipleNumbers ? (
+          <Grid item>
+            <TextField
+              style={{
+                background: 'white',
+              }}
+              multiline
+              variant="outlined"
+              label="Phone numbers"
+              value={newAllowlistMultiplePhoneNumbers}
+              onChange={handlePhoneNumberChange}
+            />
+          </Grid>
+        ) : (
+          <Grid item>
+            <TextField
+              style={{
+                background: 'white',
+              }}
+              variant="outlined"
+              label="Phone number"
+              value={newAllowlistPhoneNumber}
+              onChange={handlePhoneNumberChange}
+              InputProps={{
+                inputComponent: PhoneNumberMasks.SG as any,
+              }}
+            />
+          </Grid>
+        )}
+        <Grid item>
+          <FormControl
+            variant="outlined"
+            style={{
+              background: 'white',
+            }}
+          >
+            <InputLabel>Country</InputLabel>
+            <Select
+              native
+              onChange={(e: React.ChangeEvent<any>) =>
+                setNewAllowlistCountry(e.target.value)
+              }
+              label="Country"
+            >
+              <option value="BD">BD</option>
+              <option value="SG">SG</option>
+            </Select>
+          </FormControl>
+        </Grid>
+        <Grid item>
+          <FormControlLabel
+            control={
+              <Switch
+                name="useMultipleNumbers"
+                checked={useMultipleNumbers}
+                onChange={() => setUseMultipleNumbers(!useMultipleNumbers)}
+              />
+            }
+            label="Multiple Entries"
+          />
+        </Grid>
+
+        <Button
+          type="submit"
+          value="submit"
+          variant="contained"
+          color="primary"
+          size="small"
         >
-          <MenuItem value="BD">BD</MenuItem>
-          <MenuItem value="SG">SG</MenuItem>
-        </Select>
-      </div>
-      <PrimaryButton type="submit" value="submit">
-        Add entry
-      </PrimaryButton>
+          Add entry
+        </Button>
+      </Grid>
     </form>
   );
 
+  const rows: { id: any; phoneNumber: any; destinationCountry: any }[] = [];
+
+  allowlistEntries.map((allowlistEntry: any) =>
```

Comment:
Try defining an interface for AllowlistEntry. It would be best to put it in a `services/` file, but it would be ok in this file too.